### PR TITLE
RHTAPSRE-458: Add admission webhook

### PR DESCRIPTION
**Pull request by gbenhaim**
The admission webhook will validate the creation and update of Repository objects. It will enforce that the url of the repository matches one of the prefixes provided in a config file.

Using this admission webhook, an administrator can control which repositories can be used with pipelines-as-code.

The core logic and tests are included under the `api` package.

This project was scaffolded using kubebuilder.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	urlPrefixAllowList, err := api.LoadUrlPrefixAllowListFromFile(urlPrefixAllowListPath, os.ReadFile)
+	if err != nil {
+		setupLog.Error(err, "Failed to load url prefix allow list from file", "path", urlPrefixAllowList)
+		os.Exit(1)
+	}
+
```
Comment (ifireball) [inline cmd/main.go:140]: Did you consider putting the validator initialization inside SetupWebhookWithManager? If you did, why not do that?

**Comment by hugares** [comment]
There are about 4 kube-linter issues in the config folder. Might be an issues when you try to pull this in infra-deployments.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# repository-validator
+// TODO(user): Add simple overview of use/purpose
+
+## Description
+// TODO(user): An in-depth paragraph about your project and overview of use
+
+## Getting Started
+
+### Prerequisites
+- go version v1.21.0+
```
Comment (hugares) [inline README.md:10]: this should b 1.20 according to the docker build

**Comment by gbenhaim** [comment]
> There are about 4 kube-linter issues in the config folder. Might be an issues when you try to pull this in infra-deployments.

@hugares thanks for letting me know. Some of them are because of the directory structure that kubebuilder creates. I'm going to have another PR for creating the overlay required for deploying on OCP, I'll ensure it doesn't have the kubelinter issues.

File: README.md (hunk 2 above)
Comment (gbenhaim) [reply README.md]: fixed

**Pull request merged**